workflow-templates/im-deploy-az-app-service-manually.yml

```diff
@@ -45,70 +45,54 @@ jobs:
 
       - name: Set ENVIRONMENT
         id: clean-env
-        uses: im-open/map-input-action@v1.0.1
+        uses: im-open/set-environment-variables-by-scope@v1.0.2
         with:
-          input: ${{ github.event.inputs.environment }}
+          scope: ${{ github.event.inputs.environment }}
+          create-output-variables: true
+          error_on_no_match: true
+          custom_error_message: 'The environment must be Dev, QA, Stage Demo, UAT or Prod' # TODO:  Update for the environments that are available
+        env:
           # TODO:  Update for the environments your project contains
           # The value array contains the environments it will match and the corresponding key is
           # the environment it will output if one of the values was found.  It is case insensitive.
-          input_map: |
-            { 
-              "dev": ["dev", "d", "development"], 
-              "qa": ["qa", "q"], 
-              "stage": ["stg", "s", "stage"], 
-              "demo": ["o", "demo"], 
-              "uat": ["u", "uat"], 
-              "prod": ["prod", "production", "p"] 
-            }
-          error_on_no_match: true
-          custom_error_message: 'The environment must be Dev, QA, Stage Demo, UAT or Prod' # TODO:  Update for the environments that are available
+          mapped_input@dev d development: 'dev'
+          mapped_input@qa q: 'qa'
+          mapped_input@stg s stage: 'stage'
+          mapped_input@o demo: 'demo'
+          mapped_input@u uat: 'uat'
+          mapped_input@prod p production: 'prod'
```

Comment:
Why are these called mapped_input, can we not call it ENVIRONMENT so it's not ambiguous about what is being set?  

I also think this block should be combined with the one below.  Let's not call this action twice in one job.

Reply:
I can change the output to `ENVIRONMENT`.  I had just left it to the same out as before so nothing obvious would break.

The real problem here is we're relying on user input for this step and the scope mapping doesn't line up the same for the possible key selection.

Reply:
@hpractv  Did you see the second part of this comment about combining the actions in the set-vars job?

Reply:
After trying it out I see why it wouldn't work so well to combine the two steps in set-vars job.